pyrogram/client/methods/users/set_username.py

```diff
@@ -0,0 +1,46 @@
+# Pyrogram - Telegram MTProto API Client Library for Python
+# Copyright (C) 2017-2019 Dan Tès <https://github.com/delivrance>
+#
+# This file is part of Pyrogram.
+#
+# Pyrogram is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Pyrogram is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Pyrogram.  If not, see <http://www.gnu.org/licenses/>.
+
+from pyrogram.api import functions
+from ...ext import BaseClient
+
+
+class SetUsername(BaseClient):
+    def set_username(self,
```

Comment:
Nice feature to have, but you forget that channel/supergroup usernames can also be changed.
We need two separate methods to distinguish the cases:

- `update_user_username(username: str)` (or simply update_username): for users only
- `update_chat_username(chat_id: int or str, username: str)`: for channels/supergroups only

Methods named `update_*` will also handle the removal of usernames by passing either an empty string or None explicitly; this will be explained in the docstrings. Alternatively we can have `set_*` and `remove_*` methods which do one thing only, set/update and remove. I personally prefer the first variant.